ORStools/proc/directions_lines_proc.py

```diff
@@ -166,7 +168,7 @@ def processAlgorithm(self, parameters, context, feedback):
         return {self.OUT: dest_id}
 
     @staticmethod
-    def _get_sorted_lines(layer, field_name):
+    def _get_sorted_lines(layer: QgsProcessingParameterFeatureSource, field_name: str) -> Generator:
```

Comment:
I suspect this should be

```suggestion
    def _get_sorted_lines(layer: QgsFeatureSource, field_name: str) -> Generator:
```

Reply:
Why not a `QgsProcessingFeatureSource`?